modules/ting_search_autocomplete/includes/ting_search_autocomplete.ajax.inc

```diff
@@ -55,39 +49,5 @@ function ting_search_autocomplete_callback($query = '', $reset = FALSE) {
  * @return array
  */
 function ting_search_autocomplete_get_suggestions($query) {
-  $items = array();
-  $rest_url = variable_get('ting_search_autocomplete_suggestion_url', '');
-  $method = variable_get('ting_search_autocomplete_method', 'facets');
-  $settings = ting_search_autocomplete_settings();
-
-  // minimumString is not an openSuggestion parameter
-  unset($settings['minimumString']);
-
-  $rest_url = $rest_url . 'rest/' . $method;
-  $settings['query'] = $query;
-  $url = url($rest_url, array('query' => $settings));
-
-  $result = drupal_http_request($url, array());
-
-  if (!empty($result->data)) {
-    $result_data = drupal_json_decode($result->data);
-    if (empty($result->error)) {
-      if (!empty($result_data['suggestions'])) {
-        foreach ($result_data['suggestions'] as $suggestion) {
-          if ($method == 'facetSpell') {
-            $items[truncate_utf8(strip_tags($suggestion['phrase']), 256, TRUE, FALSE, 1)] = $suggestion['phrase'];
-          }
-          else {
-            $items[truncate_utf8(strip_tags($suggestion['suggestion']), 256, TRUE, FALSE, 1)] = $suggestion['suggestion'];
-          }
-        }
-      }
-    }
-    else {
-      watchdog('ting_search_autocomplete', 'Autocomplete error: ' . $result->error . ' (' . $result->code . ")\nURL: $url" , array(), $severity = WATCHDOG_ERROR);
-    }
-  }
-
-
-  return $items;
+  return ding_provider_invoke('search', 'autocomplete', $query);
```

Comment:
I think we can remove this function entirely. The DocBlock does not make sense either.

Reply:
Also: Remember to do an implementation of `search_autocomplete` in `connie_search`.

Reply:
You may want to pass additional general settings to the provider e.g. `maxSuggestions` which currently still belongs to `ting_search_autocomplete`.